autogpt_platform/frontend/src/components/contextual/GoogleDrivePicker/GoogleDrivePickerInput.tsx

```diff
@@ -0,0 +1,131 @@
+import { Button } from "@/components/atoms/Button/Button";
+import { cn } from "@/lib/utils";
+import { Cross2Icon } from "@radix-ui/react-icons";
+import React from "react";
+import { GoogleDrivePicker } from "./GoogleDrivePicker";
+import type { GoogleDrivePickerConfig } from "@/lib/autogpt-server-api/types";
+
+export interface GoogleDrivePickerInputProps {
+  config: GoogleDrivePickerConfig;
+  value: any;
+  onChange: (value: any) => void;
+  error?: string;
+  className?: string;
+  showRemoveButton?: boolean;
+}
+
+export function GoogleDrivePickerInput({
+  config,
+  value,
+  onChange,
+  error,
+  className,
+  showRemoveButton = true,
+}: GoogleDrivePickerInputProps) {
+  const [pickerError, setPickerError] = React.useState<string | null>(null);
+  const isMultiSelect = config.multiselect || false;
+  const currentFiles = isMultiSelect
+    ? Array.isArray(value)
+      ? value
+      : []
+    : value
+      ? [value]
+      : [];
+
+  function handlePicked(files: any[]) {
```

Comment:
this is being passed into a child component, so I'd suggest using `useCallback` here